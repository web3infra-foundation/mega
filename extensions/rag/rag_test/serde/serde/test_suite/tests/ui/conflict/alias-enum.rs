#![allow(non_camel_case_types)]

use serde_derive::Deserialize;

#[derive(Deserialize)]
enum E {
    S1 {
        #[serde(alias = "a", alias = "b", alias = "c")]
        a: (),

        // Warning on "c" and "b"
        #[serde(alias = "c")]
        b: (),

        #[serde(skip_deserializing)]
        c: (),
    },

    S2 {
        #[serde(alias = "b", alias = "c")]
        a: (),

        // Warning on "c"
        #[serde(rename = "c")]
        b: (),
    },

    #[serde(rename_all = "UPPERCASE")]
    S3 {
        #[serde(alias = "B", alias = "c")]
        a: (),

        // Warning on "b" because this collides with the "B" above after
        // applying rename rules
        b: (),
    },
}

#[derive(Deserialize)]
enum E1 {
    #[serde(alias = "a", alias = "b", alias = "c")]
    a,

    // Warning on "c" and "b"
    #[serde(alias = "c")]
    b,

    #[serde(skip_deserializing)]
    c,
}

#[derive(Deserialize)]
enum E2 {
    #[serde(alias = "b", alias = "c")]
    a,

    // Warning on "c"
    #[serde(rename = "c")]
    b,
}

#[derive(Deserialize)]
#[serde(rename_all = "UPPERCASE")]
enum E3 {
    #[serde(alias = "B", alias = "c")]
    a,

    // Warning on "b" because this collides with the "B" above after applying
    // rename rules
    b,
}

fn main() {
    __FAIL__;
}
