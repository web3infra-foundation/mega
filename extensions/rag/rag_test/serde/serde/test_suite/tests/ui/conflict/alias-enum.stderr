error[E0425]: cannot find value `__FAIL__` in this scope
  --> tests/ui/conflict/alias-enum.rs:74:5
   |
74 |     __FAIL__;
   |     ^^^^^^^^ not found in this scope

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:13:9
   |
8  |         #[serde(alias = "a", alias = "b", alias = "c")]
   |                                      --- matches all the relevant values
...
13 |         b: (),
   |         ^ no value can reach this
   |
   = note: `#[warn(unreachable_patterns)]` on by default

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:12:25
   |
8  |         #[serde(alias = "a", alias = "b", alias = "c")]
   |                                                   --- matches all the relevant values
...
12 |         #[serde(alias = "c")]
   |                         ^^^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:24:26
   |
20 |         #[serde(alias = "b", alias = "c")]
   |                                      --- matches all the relevant values
...
24 |         #[serde(rename = "c")]
   |                          ^^^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:35:9
   |
30 |         #[serde(alias = "B", alias = "c")]
   |                         --- matches all the relevant values
...
35 |         b: (),
   |         ^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:46:5
   |
41 |     #[serde(alias = "a", alias = "b", alias = "c")]
   |                                  --- matches all the relevant values
...
46 |     b,
   |     ^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:45:21
   |
41 |     #[serde(alias = "a", alias = "b", alias = "c")]
   |                                               --- matches all the relevant values
...
45 |     #[serde(alias = "c")]
   |                     ^^^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:58:22
   |
54 |     #[serde(alias = "b", alias = "c")]
   |                                  --- matches all the relevant values
...
58 |     #[serde(rename = "c")]
   |                      ^^^ no value can reach this

warning: unreachable pattern
  --> tests/ui/conflict/alias-enum.rs:70:5
   |
65 |     #[serde(alias = "B", alias = "c")]
   |                     --- matches all the relevant values
...
70 |     b,
   |     ^ no value can reach this
